test/runtime/monitor.go

```diff
@@ -249,4 +257,54 @@ var _ = Describe("RuntimeValidatedMonitorTest", func() {
 			Expect(monitorRes[2].Output().String()).Should(ContainSubstring(str))
 		}
 	})
+
+	It("checks container ids match monitor output", func() {
+		res := vm.ExecCilium(fmt.Sprintf("config %s=true %s=%s",
+			MonitorDebug, helpers.PolicyEnforcement,
+			helpers.PolicyEnforcementAlways))
+		res.ExpectSuccess()
+
+		prepareContainers()
+		defer cleanupContainers()
+
+		ctx, cancel := context.WithCancel(context.Background())
+		res = vm.ExecContext(ctx, "cilium monitor -v")
+
+		vm.ContainerExec(helpers.Client, helpers.Ping(helpers.Server))
+		vm.ContainerExec(helpers.Server, helpers.Ping(helpers.Client))
+
+		endpoints, err := vm.GetEndpointsIDMap()
+		Expect(err).Should(BeNil())
+
+		helpers.Sleep(2)
+		cancel()
+
+		// Expected full example output:
+		// CPU 01: MARK 0x3de3947b FROM 48896 DEBUG: Attempting local delivery for container id 29381 from seclabel 263
```

Comment:
Why you are not using the `ContainSubstring` function? 

You can do something like: 

```
eps := vm.GetEndpointsIds()
Expect(res).To(ContainSubstring(fmt.Sprintf("FROM %s DEBUG: Attempting local delivery for container id %s from seclabel", eps[helpers.Client], eps[helpers.Server])
```

Less logic to understand, and you're validating that the containers IDS are correct. 

Regards

Reply:
I'm converting the test over pretty faithfully, so it's not only testing that the endpoints we have are in the output, but also that the output doesn't have endpoints that don't exist. That's why it's reading the endpoint IDs out of the output and performing the lookup, rather than just validating that the expected endpoints exist in the monitor output.